src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -588,52 +611,76 @@ private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes,
             do
             {
                 AssertRead<Vector128<sbyte>>(src, srcStart, sourceLength);
-                Vector128<sbyte> str = Sse2.LoadVector128(src).AsSByte();
+                Vector128<byte> str = Vector128.LoadUnsafe(ref *src);
 
                 // lookup
-                Vector128<sbyte> hiNibbles = Sse2.And(Sse2.ShiftRightLogical(str.AsInt32(), 4).AsSByte(), mask2F);
-                Vector128<sbyte> loNibbles = Sse2.And(str, mask2F);
-                Vector128<sbyte> hi = Ssse3.Shuffle(lutHi, hiNibbles);
-                Vector128<sbyte> lo = Ssse3.Shuffle(lutLo, loNibbles);
+                Vector128<byte> hiNibbles = Vector128.ShiftRightLogical(str.AsInt32(), 4).AsByte();
+                if (Ssse3.IsSupported)
+                {
+                    // Ensure the high bits are masked off for the shuffle.
+                    hiNibbles = Vector128.BitwiseAnd(hiNibbles, mask2F);
+                }
+                Vector128<byte> hi = SimdShuffle(lutHi, hiNibbles, f);
+                Vector128<byte> lo = SimdShuffle(lutLo, str, f);
 
                 // Check for invalid input: if any "and" values from lo and hi are not zero,
                 // fall back on bytewise code to do error checking and reporting:
-                if (Sse2.MoveMask(Sse2.CompareGreaterThan(Sse2.And(lo, hi), zero)) != 0)
+                if (Vector128.BitwiseAnd(lo, hi).AsUInt64().ToScalar() != 0)
```

Comment:
This one might be better as `BitwiseAnd != Vector128<byte>.Zero` which will do an optimal comparison against zero

Reply:
-or-

```suggestion
                if ((lo & hi) != Vector128<byte>.Zero)
```